From 6d1c318d9bf7da0a0513469edd2a521c0d8e9ee0 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Fri, 10 Aug 2018 01:12:22 +0200
Subject: [PATCH 21/24] Modify color inversion matrix to swap Red and Blue
 colors

Instead of color inversion it will produce Red and Blue colors swap,
in order to correct displayed colors for R4xx and other old drivers
still based on KMS API
---
 .../server/accessibility/DisplayAdjustmentUtils.java      | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/services/accessibility/java/com/android/server/accessibility/DisplayAdjustmentUtils.java b/services/accessibility/java/com/android/server/accessibility/DisplayAdjustmentUtils.java
index c81a876fe39e..f97d7673936e 100644
--- a/services/accessibility/java/com/android/server/accessibility/DisplayAdjustmentUtils.java
+++ b/services/accessibility/java/com/android/server/accessibility/DisplayAdjustmentUtils.java
@@ -50,10 +50,10 @@ class DisplayAdjustmentUtils {
      * for full implementation details.
      */
     private static final float[] MATRIX_INVERT_COLOR = new float[] {
-        0.402f, -0.598f, -0.599f, 0,
-       -1.174f, -0.174f, -1.175f, 0,
-       -0.228f, -0.228f,  0.772f, 0,
-             1,       1,       1, 1
+       0.0f, 0.0f, 1.0f, 0,
+       0.0f, 1.0f, 0.0f, 0,
+       1.0f, 0.0f, 0.0f, 0,
+          0,    0,    0, 1
     };
 
     public static void applyDaltonizerSetting(Context context, int userId) {
-- 
2.17.1

